Returning nothing is an implicit return of zero, which means we failed
rather than succeeded if the fragment table is empty. Backport two
patches from CVS to properly return TRUE in that case.

Index: unsquash-3.c
===================================================================
RCS file: /cvsroot/squashfs/squashfs/squashfs-tools/unsquash-3.c,v
retrieving revision 1.10
retrieving revision 1.11
diff -u -p -r1.10 -r1.11
--- a/squashfs-toosl/unsquash-3.c	31 Mar 2009 04:38:53 -0000	1.10
+++ b/squashfs-tools/unsquash-3.c	21 Apr 2009 00:58:22 -0000	1.11
@@ -36,7 +36,7 @@ int read_fragment_table_3()
 		sBlk.fragment_table_start);
 
 	if(sBlk.fragments == 0)
-		return;
+		return TRUE;
 
 	if((fragment_table = malloc(sBlk.fragments *
 			sizeof(squashfs_fragment_entry_3))) == NULL)
Index: unsquash-4.c
===================================================================
RCS file: /cvsroot/squashfs/squashfs/squashfs-tools/unsquash-4.c,v
retrieving revision 1.20
retrieving revision 1.21
diff -u -p -r1.20 -r1.21
--- a/squashfs-tools/unsquash-4.c	31 Mar 2009 04:38:53 -0000	1.20
+++ b/squashfs-tools/unsquash-4.c	21 Apr 2009 00:59:16 -0000	1.21
@@ -38,7 +38,7 @@ int read_fragment_table_4()
 		sBlk.fragment_table_start);
 
 	if(sBlk.fragments == 0)
-		return;
+		return TRUE;
 
 	if((fragment_table = malloc(sBlk.fragments *
 			sizeof(squashfs_fragment_entry))) == NULL)
